src/main/scala/is/hail/table/Table.scala

```diff
@@ -1138,4 +1139,95 @@ class Table(val hc: HailContext, val tir: TableIR) {
     assert(hintPartitioner.forall(p => p.pkType.types.sameElements(orderedKTType.pkType.types)))
     OrderedRVD.coerce(orderedKTType, rvd, None, hintPartitioner)
   }
+
+  def intervalJoin(other: Table, fieldName: String): Table = {
+    assert(other.keySignature.exists(s => s.size == 1 && s.types(0).isInstanceOf[TInterval]))
+    val intervalType = other.keySignature.get.types(0).asInstanceOf[TInterval]
+    assert(keySignature.exists(s => s.size == 1 && s.types(0) == intervalType.pointType))
+
+    val leftORVD = rvd match {
+      case ordered: OrderedRVD => ordered
+      case unordered =>
+        OrderedRVD.coerce(
+          new OrderedRVDType(key.get.toArray, key.get.toArray, signature),
+          unordered)
+    }
+
```

Comment:
extra space

Reply:
This entire section was copied and modified from MatrixTable.annotateRowsIntervalTable. I didn't want to take the time to abstract out the common pieces from the different ones, especially since Patrick is going to delete all of this when his join stuff is ready.

Reply:
I did address your comments though (and apply them to both places)